### ISSUE TITLE
Copy should retain Content-Type from source

### ISSUE THREAD

**Issue by sskylar** [ncw/rclone#733]
Using `rclone copy` all files appeared to transfer correctly from my source to destination (nice!).

However, on closer inspection I found rclone lost/changed some of my mime types. 

When copying, I think it would be ideal to retain the `Content-Type` (mime) from the source (wherever possible) rather than [guessing](https://github.com/ncw/rclone/blob/83ba59749f4fd1b5d0ab558dd7ff603f81d7fec2/fs/operations.go#L172) from the file extension. Guessing works decently well for most files, but I believe this should only be done when absolutely necessary, as fallback for sources that may not provide a explicit mime.

For example on Rackspace Cloud Files I have a file called `image-1` (no extension) with `Content-Type: image/jpeg`:

```
$ curl -I http://xxx.cf2.rackcdn.com/image-1
HTTP/1.1 200 OK
ETag: 9a9d9f5229c256babf8e80707747e294
Content-Length: 49583
Content-Type: image/jpeg
```

Using rclone I'm able to copy this perfectly to S3, except the `Content-Type` is lost:

```
$ curl -I http://xxx.s3.amazonaws.com/image-1
HTTP/1.1 200 OK
ETag: "9a9d9f5229c256babf8e80707747e294"
Content-Length: 49583
Content-Type: application/octet-stream
```

Instead of `application/octet-stream` it should be `image/jpeg`

--

> What is your rclone version (eg output from `rclone -V`)

1.33

> Which OS you are using and how many bits (eg Windows 7, 64 bit)

macOS 10.12, 64 bit 

> Which cloud storage system are you using? (eg Google Drive)

Rackspace Cloud Files (source) and Amazon S3 (destination)

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone copy -v rackspace:xxx s3:xxx`

> A log from the command with the `-v` flag (eg output from `rclone -v copy /tmp remote:tmp`)

```
Transferred:   18.254 MBytes (681.081 kBytes/s)
Errors:                 0
Checks:                 8
Transferred:           17
Elapsed time:       27.4s
2016/09/17 15:29:28 Go routines at exit 5
```

**Comment by DurvalMenezes** [comment]
I might be wrong, but I do not think that rclone has anything to do with that. My reasoning is that the content-type for a file sitting in a filesystem will generally (with few exceptions for some filesystems) do not have any "content-type" metadata associated with them. This is the case, for example, with the extN filesystems running in Linux, and with FAT and NTFS running on Windows: to figure out the content-type, Linux programs (eg, "file") look at the file data for signatures, and Windows guesses based on the file extension. I know Macs are (were?) different, with each file having a "data fork" for its data and a "resource fork" for metadata (presumably including content-type), but this is the exception, not the rule. 

In summary: rclone transfers the *data* for the file, and a very limited set of the file's metadata (basically just the file name and perhaps the timestamp, but nothing much more). It will almost certainly not try to transfer a content-type even if one is available (eg, Macs) even if the remote cloud storage supports it.  Therefore, what is happening in your case is that the Amazon HTTP server is guessing the content-type wrong (probably because the file has no extension). To prove that, just transfer the file to a cloud filename ending in .JPG and repeat the CURL test: I bet Amazon's HTTP server will return an "image/jpeg" for the content-type. 

Hope that helps.

**Comment by ncw** [comment]
Good explanation by @DurvalMenezes though it isn't Amazon guessing the content type from the name, it is rclone (via the go runtime). 

I could add an interface to the object which could return the content type. The local fs would carry on working like it does (or maybe on OS X it would read the content type somehow from the OS) - the others will know the content type already so as return it. This would fix the original issue. I could then list content typed in listings etc.

**Comment by ncw** [comment]
I made a little test to see how it might work: http://pub.rclone.org/v1.33-28-gdb2307f-content-type%CE%B2/

Let me know if it does!

**Comment by sskylar** [comment]
Thanks @ncw! 

Tested it out in a couple scenarios using `v1.33-28-gdb2307f-content-typeβ`, but unfortunately it looks like the content-type is being reset. In the case of files without an extension, the result is still  `application/octet-stream`.

- Rackspace -> S3 - content-type was lost
- S3 -> Rackspace - content-type was lost
- S3 -> S3 (server side copy) - content-type is retained correctly, however this already worked on `v1.33` when doing a server side copy (switched back to confirm)

Let me know if I can help debug in any way.

**Comment by ncw** [comment]
Ah, and the code looked so good ;-) 

I'll have another go and post another beta tomorrow if I have time. 

Thanks for testing.

**Comment by ncw** [comment]
This should work a bit better...

http://pub.rclone.org/v1.33-28-g2dee7bc-content-type%CE%B2/

**Comment by ncw** [comment]
I've merged this to master - please re-open if for some reason it doesn't work for you:

http://pub.rclone.org/v1.33-30-g945f49a%CE%B2/

**Comment by seabre** [comment]
I'm on v1.35 and getting the same issue when transferring from one Rackspace account to another.

I have a file in a Cloud Files container in my Rackspace A account with a content type of image/jpeg, but when it's transferred to Clould Files container in Rackspace B, it has a content type of application/octet-stream.

The _difference_ is I'm using `sync` instead, so I'm running a command like: `rclone sync rackspaceA:xxx rackspaceB:xxx`.

**Comment by seabre** [comment]
Also, more info, I had this issue on rclone-v1.35-linux-amd64.

*However* I tried rclone-v1.34-linux-amd64 and it appears to be working.

I tried both sync *and* copy in both builds.

Maybe reopen as a regression @ncw?

**Comment by ncw** [comment]
@seabre 

I've fixed this in 

http://beta.rclone.org/v1.35-155-g10e532b/ (uploaded in 15 - 30 mins)

Thanks for reporting

**Comment by shriDeveloper** [comment]
hey @ncw  , i am having the same issue

**Comment by ncw** [comment]
@shriDeveloper can you make a new issue please?
